### PR TITLE
APP-184: [C++] Basic level Apps: Endpoint profiles and grouping

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @slash32 @Dubland

### CODE REVIEW

**Comment by Oldnickus** [comment]
@kaa-leeroy test this please

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

**Comment by Oldnickus** [comment]
@kaa-leeroy add to whitelist

File: profilingdemo/source/cpp/src/KaaDemo.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        return profile_;
+    }
+
+private:
+    KaaProfile profile_;
+};
+
+const std::vector<ProfileData> clientProfiles = {
+    ProfileData(true, false, true),
```
Comment (vadimol) [inline profilingdemo/source/cpp/src/KaaDemo.cpp]: static

File: profilingdemo/source/cpp/src/KaaDemo.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    const KaaProfile &getProfile() const
+    {
+        return profile_;
+    }
+
+private:
+    KaaProfile profile_;
+};
+
+const std::vector<ProfileData> clientProfiles = {
```
Comment (vadimol) [inline profilingdemo/source/cpp/src/KaaDemo.cpp]: static

File: profilingdemo/source/cpp/src/KaaDemo.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    ~KaaClientManager()
+    {
+        for (auto &client : kaaClients_) {
+            client->stop();
+        }
+    }
+
+    void spawnKaaClient(const KaaProfile &profile)
+    {
+        std::string clientDir = "client"+std::to_string(kaaClients_.size());
```
Comment (vadimol) [inline profilingdemo/source/cpp/src/KaaDemo.cpp]: code style

File: profilingdemo/source/cpp/build.sh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            exit 1
+        fi
+
+        mkdir -p $KAA_LIB_PATH &&
+        tar -zxf $KAA_SDK_TAR_NAME -C $KAA_LIB_PATH
+    fi
+}
+
+function build_app {
+    cd $PROJECT_HOME &&
```
Comment (vadimol) [inline profilingdemo/source/cpp/build.sh]: please, add quotes

File: profilingdemo/source/cpp/src/KaaDemo.cpp [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+};
+
+const std::vector<ProfileData> clientProfiles = {
+    ProfileData(true, false, true),
+    ProfileData(false, false, true),
+    ProfileData(true, true, true),
+    ProfileData(true, true, false),
+};
+
+// TODO: move this typedef to SDK.
```
Comment (vadimol) [inline profilingdemo/source/cpp/src/KaaDemo.cpp]: TODO without Jira

**Comment by vadimol** [comment]
Few minor comments, otherwise looks good.

**Comment by vadimol** [comment]
Looks good for me

**Pull request merged**